### ISSUE TITLE
TypeError during audio download / post-processing

### ISSUE THREAD

**Issue by yyakko** [yt-dlp/yt-dlp#10339]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Provide a description that is worded well enough to be understood

After upgrading to version 2024.07.01, yt-dlp gives the following error when trying to download audio from youtube:

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/local/bin/yt-dlp/__main__.py", line 17, in <module>
  File "/usr/local/bin/yt-dlp/yt_dlp/__init__.py", line 1081, in main
  File "/usr/local/bin/yt-dlp/yt_dlp/__init__.py", line 1071, in _real_main
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3603, in download
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3578, in wrapper
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1611, in extract_info
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1622, in wrapper
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1778, in __extract_info
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1837, in process_ie_result
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3012, in process_video_result
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 179, in wrapper
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3557, in process_info
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3739, in post_process
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3721, in run_all_pps
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3699, in run_pp
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/common.py", line 23, in run
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/common.py", line 128, in wrapper
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/ffmpeg.py", line 884, in run
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/ffmpeg.py", line 862, in _fixup
  File "/usr/local/bin/yt-dlp/yt_dlp/utils/_utils.py", line 2094, in _change_extension
TypeError: validate_options.<locals>.<lambda>() got an unexpected keyword argument 'prepend'
```

The original audio file is downloaded, and the error seems to occur during post-processing or adding metadata. 
The problem disappeared after downgrading to version 2024.05.27.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '-x', '--audio-format', 'opus', '--audio-quality', '24k', '-f', 'ba[ext=m4a]', '--ppa', 'ExtractAudio:-af volume=5dB', '--add-metadata', '--parse-metadata', 'uploader:%(album)s', 'https://www.youtube.com/watch?v=VRmBZXAwOv8', '-o', '%(upload_date)s_%(uploader)s_%(title)s.%(ext)s']
[debug] User config "~/.config/yt-dlp/config": ['--compat-options', 'all', '-f', 'bv*[vcodec^=avc1][height=720]+ba[ext=m4a]/bv*[height=720]+ba/b[height=720]/b*', '--live-from-start', '--concurrent-fragments', '3', '--geo-bypass', '--sponsorblock-remove', 'sponsor', '--merge-output-format', 'mp4', '-o', '%(upload_date)s_%(uploader)s_%(title)s.%(ext)s']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.07.01 from yt-dlp/yt-dlp [5ce582448] (zip)
[debug] Compatibility options: list-formats, no-external-downloader-progress, prefer-legacy-http-handler, no-youtube-channel-redirect, abort-on-error, no-direct-merge, multistreams, format-spec, no-playlist-metafiles, no-clean-infojson, allow-unsafe-ext, no-live-chat, no-certifi, embed-thumbnail-atomicparsley, no-keep-subs, format-sort, no-youtube-prefer-utc-upload-date, playlist-index, embed-metadata, manifest-filesize-approx, no-attach-info-json, filename-sanitization, playlist-match-filter, seperate-video-versions, *filename, no-youtube-unavailable-videos
[debug] Python 3.11.2 (CPython x86_64 64bit) - Linux-6.1.0-22-amd64-x86_64-with-glibc2.36 (OpenSSL 3.0.13 30 Jan 2024, glibc 2.36)
[debug] exe versions: ffmpeg 6.0 (fdk,setts), ffprobe 6.0, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, pyxattr-0.8.1, requests-2.28.1, sqlite3-3.40.1, urllib3-1.26.12, websockets-10.4
[debug] Proxy map: {}
[debug] Request Handlers: urllib
WARNING: Using allow-unsafe-ext opens you up to potential attacks. Use with great care!
[debug] Loaded 1831 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.07.01 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.07.01 from yt-dlp/yt-dlp)
[youtube] Extracting URL: https://www.youtube.com/watch?v=VRmBZXAwOv8
[youtube] VRmBZXAwOv8: Downloading webpage
[youtube] VRmBZXAwOv8: Downloading ios player API JSON
[debug] Loading youtube-nsig.71059424 from cache
[debug] [youtube] Decrypted nsig v6AE7yEhsDeFRpJhd => WNoSFYijZLHPEA
[youtube] VRmBZXAwOv8: Downloading m3u8 information
[debug] Sort order given by user: hasaud, lang, quality, tbr, filesize, vbr, height, width, proto, vext, abr, aext, fps, fs_approx, source, id
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, hasaud, lang, quality, tbr, filesize, vbr, height, width, proto, vext, abr, aext, fps, fs_approx, source, id, res, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr
[debug] Searching for '(?P<album>.+)' in '%(uploader)s'
[MetadataParser] Parsed album from '%(uploader)s': 'Bent Outta Shape Chess'
[info] VRmBZXAwOv8: Downloading 1 format(s): 140
[debug] Invoking http downloader on "https://rr12---sn-n8v7sney.googlevideo.com/videoplayback?expire=1719978994&ei=kneEZv7pH8jXv_IPj_WEuAc&ip=95.24.32.21&id=o-AF9iotLcLkPEMYI8YEWG8BeXUl8P_w8IpbKSTYShHhFi&itag=140&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=nm&mm=31%2C26&mn=sn-n8v7sney%2Csn-c0q7lnz7&ms=au%2Conr&mv=m&mvi=12&pl=22&initcwndbps=2060000&vprv=1&svpuc=1&mime=audio%2Fmp4&rqh=1&gir=yes&clen=27311247&dur=1687.510&lmt=1573129544173294&mt=1719956956&fvip=5&keepalive=yes&c=IOS&txp=1301222&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cvprv%2Csvpuc%2Cmime%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRAIgP1V7_cy70A7XdWuVPh9ZhfQ6YKi6D2eG7E6_3YkIXW8CIF-14YizmbxQaXsAo1UgGnmfzMuD-Tvc4_ca_5ik0oE_&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AHlkHjAwRQIhAKF7KmZr1zNOkyyPQKnEqm1_C26KnOeCvRmbFUY3tbevAiBg800rMpdtISuBMoSX9rHWZSV3S5xWV_bR6Pb31nMHpA%3D%3D"
[download] Destination: 20120402_Bent Outta Shape Chess_Thomas Hobbes.m4a
[download] 100% of   26.05MiB in 00:00:14 at 1.81MiB/s
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/local/bin/yt-dlp/__main__.py", line 17, in <module>
  File "/usr/local/bin/yt-dlp/yt_dlp/__init__.py", line 1081, in main
  File "/usr/local/bin/yt-dlp/yt_dlp/__init__.py", line 1071, in _real_main
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3603, in download
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3578, in wrapper
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1611, in extract_info
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1622, in wrapper
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1778, in __extract_info
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1837, in process_ie_result
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3012, in process_video_result
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 179, in wrapper
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3557, in process_info
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3739, in post_process
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3721, in run_all_pps
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3699, in run_pp
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/common.py", line 23, in run
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/common.py", line 128, in wrapper
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/ffmpeg.py", line 884, in run
  File "/usr/local/bin/yt-dlp/yt_dlp/postprocessor/ffmpeg.py", line 862, in _fixup
  File "/usr/local/bin/yt-dlp/yt_dlp/utils/_utils.py", line 2094, in _change_extension
TypeError: validate_options.<locals>.<lambda>() got an unexpected keyword argument 'prepend'
```

**Comment by dirkf** [comment]
This looks like the problem fixed in https://github.com/yt-dlp/yt-dlp/pull/10336, already merged. See https://github.com/yt-dlp/yt-dlp/commit/5ce582448ececb8d9c30c8c31f58330090ced03a#commitcomment-143745778.

**Comment by bashonly** [comment]
From the readme:

> For ease of use, a few more compat options are available:
>
> * `--compat-options all`: Use all compat options (**Do NOT use**)

You are opening yourself up to multiple security vulnerabilities by having that in your config.

Fixed in 773bbb181506856ffda95496ab60c1c9603f1f71

New release shortly
